### PR TITLE
Expose dialect-supported types (#181)

### PR DESCRIPTION
**Pull request by jcampbell**
## Todos
- [x] MIT compatible
- [x] Tests
- [x] Documentation
- [x] Updated CHANGES.rst

### CODE REVIEW

File: sqlalchemy_redshift/dialect.py [hunk 1]
```diff
@@ -135,6 +136,32 @@ class RedshiftImpl(postgresql.PostgresqlImpl):
 ])
 
 
+class TIMESTAMPTZ(sa.dialects.postgresql.TIMESTAMP):
+
+    __visit_name__ = 'TIMESTAMPTZ'
+
+    def __init__(self):
+        super(TIMESTAMPTZ, self).__init__(timezone=True)
```
Comment (jklukas) [inline sqlalchemy_redshift/dialect.py]: Can you add a docstring here explaining why this is necessary? I assume TIMESTAMPTZ is a Redshift-specific datatype name?

File: tests/test_dialect_types.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import sqlalchemy_redshift.dialect
+import sqlalchemy
+
+def test_defined_types():
+    # These types are the officially supported Redshift types as of 20191121
```
Comment (jklukas) [inline tests/test_dialect_types.py]: Can you add a link to Redshift docs page where this list is documented?

Comment (jcampbell) [reply tests/test_dialect_types.py]: Added!

File: sqlalchemy_redshift/dialect.py (hunk 1 above)
Comment (jcampbell) [reply sqlalchemy_redshift/dialect.py]: Added!

**Comment by jcampbell** [comment]
I wasn't able to address one of the linting requests because the requested link URL itself is longer than the character requirement.

File: sqlalchemy_redshift/dialect.py [hunk 3]
```diff
@@ -135,6 +154,24 @@ class RedshiftImpl(postgresql.PostgresqlImpl):
...
+    """
+    Redshift defines a TIMTESTAMPTZ column type as an alias
+    of TIMESTAMP WITH TIME ZONE.
+    https://docs.aws.amazon.com/redshift/latest/dg/c_Supported_data_types.html
+
+    Adding an explicit type to the RedshiftDialect allows us follow the
+    SqlAlchemy conventions for "vendor-specific types."
+
+    https://docs.sqlalchemy.org/en/13/core/type_basics.html#vendor-specific-types
+    """
```
Comment (jklukas) [inline sqlalchemy_redshift/dialect.py]:
```suggestion
    """ # noqa
```

Adding a noqa comment here should satisfy the linter.

**Comment by jcampbell** [comment]
@jklukas : I rebased to recombine to a single commit for your updated review!

**Comment by jcampbell** [comment]
> I wasn't able to address one of the linting requests because the requested link URL itself is longer than the character requirement.

Turns out the linter was smart...doesn't mind.

**Pull request merged**